packages/valory/contracts/mech/contract.py

```diff
@@ -138,9 +245,16 @@ def process_request_event(
         :param tx_hash: the hash of a request tx to be processed.
         :return: a dictionary with the request id.
         """
-        return cls._process_event(
-            ledger_api, contract_address, tx_hash, "Request", "requestId", "data"
-        )
+        res = {}
+        for abi in partial_abis:
+            contract_instance = ledger_api.api.eth.contract(contract_address, abi=abi)
+            res = cls._process_event(
+                ledger_api, contract_instance, tx_hash, "Request", "requestId", "data"
+            )
+            if "error" not in res:
+                return res
+
+        return res
```

Comment:
A `process_mech_event` can be extracted here, accepting the type of the request as an argument, to avoid code dup.

Reply:
#217